proxy/http/HttpSM.h

```diff
@@ -679,10 +622,58 @@ HttpTransact::State::state_machine_id() const
   return state_machine->sm_id;
 }
 
+inline bool
+HttpSM::get_client_tcp_reused() const
+{
+  return _ua.get_client_tcp_reused();
+}
+
+inline bool
+HttpSM::get_client_ssl_reused() const
+{
+  return _ua.get_client_ssl_reused();
+}
+
+inline bool
+HttpSM::get_client_connection_is_ssl() const
+{
+  return _ua.get_client_connection_is_ssl();
+}
+
+inline char const *
+HttpSM::get_client_protocol() const
+{
+  return _ua.get_client_protocol();
+}
+
+inline char const *
+HttpSM::get_client_sec_protocol() const
+{
+  return _ua.get_client_sec_protocol();
+}
+
+inline char const *
+HttpSM::get_client_cipher_suite() const
+{
+  return _ua.get_client_cipher_suite();
+}
+
+inline char const *
+HttpSM::get_client_curve() const
+{
+  return _ua.get_client_curve();
+}
+
+inline int
+HttpSM::get_client_alpn_id() const
+{
+  return _ua.get_client_alpn_id();
+}
+
```

Comment:
So, we have to add a function to `HttpSM` when we add a function to `HttpUserAgent`? Is it a bad idea to have `HttpSM::get_client` or `HttpSM::get_useragent` so we don't have to have these?

Reply:
I was asking myself this question as well. Since the number of getters here is slightly getting out of hand, I think we should go with your suggestion. The goal is to be able to get rid of many of these getters, but that change will also be easier to handle if we don't have to remove getters in 2 places every time.